scipy/signal/filter_design.py

```diff
@@ -2212,8 +2212,9 @@ def iirdesign(wp, ws, gpass, gstop, analog=False, ftype='ellip', output='ba',
                          "the same shape, got %s and %s"
                          % (wp.shape, ws.shape))
     if wp.shape[0] == 2:
-        if wp[0] < 0 or ws[0] < 0:
-            raise ValueError("Values for wp, ws can't be negative")
+        if not (0 <= wp[0] <= 1) or not (0 <= wp[1] <= 1) or \
```

Comment:
It may be more user-friendly to keep the original check, and add a new one for values above the Nyquist frequency.

_"Values for wp, ws need to be from 0 to 1"_ can be hard to interpret, compared to "can't be negative" and "is above ...."

Reply:
And then the tests need a tweak as well.